src/NodeServices/Interfaces.cs

```diff
@@ -188,4 +191,108 @@ internal interface IExtensionApplication
         /// <param name="baseDirectory"></param>
         void PreloadAsmLibraries(string baseDirectory);
     }
+
+    //TODO should this live in DynamoServices?
+    //make these internal somehow?
```

Comment:
We need this to be public, right? Otherwise only we could add completion providers.

Reply:
well... yes, but are we happy with this API enough to make it public and keep it around forever? I guess I'd rather switch to it public in the future instead - thoughts?

Reply:
I've made these all internal for now...